### ISSUE TITLE
Upgrade Chain Tutorial Should Link to Update Polkadot JS Types

### ISSUE THREAD

**Issue by shawntabrizi** [substrate-developer-hub/substrate-developer-hub.github.io#738]
This tutorial: https://substrate.dev/docs/en/tutorials/upgrade-a-chain/sudo-upgrade

Uses Node Template + Polkadot JS

But doesnt ask the user to update their types, thus they run into this error: https://polkadot.js.org/docs/api/FAQ/#the-node-returns-a-could-not-convert-error-on-send

**Comment by danforbes** [comment]
I believe that Polkadot-JS Apps UI handles this automatically for the Node Template (ref: https://github.com/polkadot-js/apps/blob/master/packages/apps-config/src/api/spec/node-template.ts) @jacogr can you please confirm?

**Comment by jacogr** [comment]
The apps UI (apart from some chains like Acala, Polkadot, Kusama) does not handle different specVersion types.

**Comment by riusricardo** [comment]
@danforbes I suggest to update this tutorial and use the front-end-template. As Jaco pointed out this is a case where even if Polkaddot JS Apps has a file to handle node-template types, it won't always works because it follows master.
The node-template `spec version` is always 0, we previously discussed this issue and came to the conclusion that it was better to keep it this way because it's a template. This is the reason why you updated the tutorials to use the front-end template as it will be in sync with the latest node-template version.

**Comment by jacogr** [comment]
There could be an issue due to 2.3.1 (fixed in the 2.4.1 API release tomorrow) where in some cases the types (as originally specified) is not always applied when the specVersion changes, upgrades or historic query related. So the above error report could very well be a bug.

**Comment by riusricardo** [comment]
@jacogr I'm not quite sure if you are referring to my point about `specVersion` is `0`. That was my mistake, it is always `1`. But my main point is to highlight what you said. That we aren't changing that number so Polkadot JS Apps can't identify the difference between node-templates.

**Comment by jacogr** [comment]
No, not refering to your comment specifically, just mentioning that there could be reports due to a funny bug that was/is existing where it breaks on upgrade and is "magically" working after a refresh. (As mentioned, this is fixed in master and due for release on Mon). I def. saw some weird/unexplained reports over the last week all of a sudden. 

As for versioning, 100% with you and understand our rationale for keeping it fixed. However if we do swap to MultiAddress for node-template (assuming that change does goes in), would actually love a bump to 2 (or something round like "10") since there may be some chaos that we can minimize on both this repo and elsewhere.

I also chatted to Shawn that I'm thinking of swapping the default to `Address: 'AccountId'` in the API since the vast majority follows that approach. (Generally follow Substrate node for defaults, but it is probably wise in this case not to)

**Comment by danforbes** [comment]
I think it's likely than any reports of unexpected behavior are related to the bug that Jaco is referring to. I definitely completed the steps in this tutorial multiple times in order to write it, so I'm certain that it worked up until very recently at least. I do think it would be a good idea to rewrite this tutorial on top of the Front-End Template, but it would require some changes to the way the Front-End Template handles `sudo.uncheckedWeight`, I think.

**Comment by riusricardo** [comment]
Yeah, I know that you tested it at the time of the release. But as I said, the JS API changes with master so it isn't stable to use it in the tutorials.
The front-end template already supports the `sudo.sudoUncheckedWeight` and the forkless upgrades. The tutorial could explain the different sudo types based in Polkadot JS API. By doing this they will get the idea for whenever they try to use it in Polkadot JS Apps and the front end template.
